src/main/java/core/basesyntax/MyHashMap.java

```diff
@@ -1,19 +1,131 @@
 package core.basesyntax;
 
+import java.util.Objects;
+
 public class MyHashMap<K, V> implements MyMap<K, V> {
+    private static final int DEFAULT_CAPACITY = 16;
+    private static final float DEFAULT_LOAD_FACTOR = 0.75f;
+    private static final int MAXIMUM_CAPACITY = 1 << 30;
+    private Node<K,V>[] table;
+    private int threshold;
+    private int size;
 
     @Override
     public void put(K key, V value) {
+        if (table == null || table.length == 0) {
+            table = resize();
+        }
+
+        int hash = keyCheck(key);
+
+        int index = hash & (table.length - 1);
 
+        for (Node<K, V> e = table[index]; e != null; e = e.next) {
+            if ((e.hash == hash) && ((e.key == null && key == null)
+                    || (e.key != null && e.key.equals(key)))) {
+                e.value = value;
+                return;
+            }
+        }
+
+        Node<K, V> newNode = table[index];
+        table[index] = new Node<>(hash, key, value, newNode);
+        size++;
+        if (size > threshold) {
+            resize();
+        }
     }
 
-    @Override
     public V getValue(K key) {
+        if (table == null) {
+            return null;
+        }
+
+        int hash = keyCheck(key);
+
+        int index = hash & (table.length - 1);
+
+        for (Node<K, V> e = table[index]; e != null; e = e.next) {
+            if ((e.hash == hash) && (Objects.equals(e.key, key))) {
+                return e.value;
+            }
+        }
+
         return null;
     }
 
     @Override
     public int getSize() {
-        return 0;
+        return size;
+    }
+
+    private Node<K,V>[] resize() {
```

Comment:
Huge method, lots of responsibilities and some `if`s are not even necessary, we have to be mindful when writing code to make it understandable for other colleagues and more, there is also a proverb: "good code can be read and understood even by a non-programmer". 
Maximum capacity is good to have of course, but for this task it was unnecessary, especially considering that you were late to send it on time.
It is often easy to dig down too deep, because it is interesting or you want to do it right, but there is no time for it, so we have to be aligned with the priorities and deadlines set by the lead, same goes for the real work.